File: MyTest/app/src/main/java/com/example/user/mytest/MainActivity.java
package com.example.user.mytest;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import android.content.Intent;
import android.view.View;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.overdraw);
    }
    public void onStartClick(View view){
        Intent intent = new Intent(this, kat_OverdrawActivity.class);
        startService(intent);
    }

    public void onEndClick(View view){
        Intent intent = new Intent(this, kat_OverdrawActivity.class);
        stopService(intent);
    }

}

File: Memo/app/src/main/java/com/example/memo/MainActivity.kt
package com.example.memo

import android.annotation.SuppressLint
import android.os.AsyncTask
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    lateinit var db : MemoDatabase
    var memoList = listOf<MemoEntity>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        db = MemoDatabase.getInstance(this)!!

        val button_add : Button = findViewById(R.id.button_add)
        val edittext_memo : EditText = findViewById((R.id.edittext_memo))
        button_add.setOnClickListener {
            val memo = MemoEntity(null,edittext_memo.text.toString())
            insertMemo(memo)


        }




    }

    //1. Insert Data
    //2. Get Data
    //3. Delete Date
    //4. Set Recycler View
    @SuppressLint("StaticFieldLeak")
    fun insertMemo(memo : MemoEntity){
        //1. MinThread vs WorkerThread(Background Thread)

        val insertTask = object : AsyncTask<Unit, Unit, Unit>(){
            override fun doInBackground(vararg params: Unit?) {
                db.MemoDAO().insert(memo)
            }

            override fun onPostExecute(result: Unit?) {
                super.onPostExecute(result)
                getAllMemos()
            }

        }
        insertTask.execute()
    }
    @SuppressLint("StaticFieldLeak")
    fun getAllMemos(){
        val getTask = object : AsyncTask<Unit,Unit,Unit>(){
            override fun doInBackground(vararg params: Unit?) {
                memoList = db.MemoDAO().getAll()
            }

            override fun onPostExecute(result: Unit?) {
                super.onPostExecute(result)
                setRecyclerView(memoList)
            }
        }

        getTask.execute()
    }
    fun deleteMemo(){

    }
    fun setRecyclerView(memoList : List<MemoEntity>){

    }

}


